pkg/controller/bastion/actuator_delete.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bastion
+
+import (
+	"context"
+	"time"
+
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	awsclient "github.com/gardener/gardener-extension-provider-aws/pkg/aws/client"
+	ctrlerror "github.com/gardener/gardener/extensions/pkg/controller/error"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/extensions"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (a *actuator) Delete(ctx context.Context, bastion *extensionsv1alpha1.Bastion) error {
+	logger := a.logger.WithValues("bastion", client.ObjectKeyFromObject(bastion), "operation", "delete")
+
+	cluster, err := extensions.GetCluster(ctx, a.Client(), bastion.Namespace)
```

Comment:
Can you rather pass the cluster as argument to the `Delete` (and `Reconcile`) functions? This is also how the `Actuator` interfaces of other extension resources look like: https://github.com/gardener/gardener/blob/c05c128cc126ccecb793cd5c72c72f93e8d0b5be/extensions/pkg/controller/controlplane/actuator.go#L26-L30